packages/core/package.json

```diff
@@ -17,12 +17,12 @@
     "e2e:stop": "make stop",
     "e2e:clean": "make stop clean",
     "e2e:test": "vitest --dangerouslyIgnoreUnhandledErrors -c vitest.e2e.config.ts --test-timeout=30000",
-    "watch:cjs": "tsc -w -p tsconfig/tsconfig.cjs.json",
-    "prepare": "npm run build"
+    "watch:cjs": "tsc -w -p tsconfig/tsconfig.cjs.json"
   },
   "author": "",
   "license": "ISC",
   "dependencies": {
```

Comment:
Thoughts on using `peerDependencies` instead of `dependencies`?

Reply:
tbh I'm just a noob when it comes to this, what should be marked as a peer dependency?

Reply:
## Explanation

Using installation docs as an example, if using `dependencies`, user will only need to install sdk itself since it already ships with required dependencies:

```bash
npm install @skip-router/core
```

- Pros: single package installation, no need to add others since it's bundled
- Cons: prone to version mismatch if project has same deps installed with different semver, and bigger deps size

If using `peerDependencies`, user need to install required dependencies alongside in their project:

```bash
npm install @skip-router/core cosmjs-types viem # and others...
```

- Pros: users can specific what version of the package is, less errors on version mismatch, and smaller deps size
- Cons: users must installed required packages to ensure sdk works as intended

## Example

@graz-sh/types has two peer deps which is also marked as optional, meaning the package won't warn if peer deps is not installed.

https://github.com/graz-sh/types/blob/main/package.json#L31-L42

## Answer

My 2¢ on which should be marked as peer deps:

- viem
- chain-registry (or just straight up drop it)
- cosmjs related deps (unless we lock it to specific version, then leave it as is)